File: kumoh-code-2017/Java/#1 기본/#1 과제/#5 java 과제/ppt 초본/은행관리/Deposit.java
public class Deposit extends Transaction
{
	private final static int CANCELED = 0;
	private int amount;
	private Keypad keypad;
	
	public Deposit(int account_number, Screen screen, BankDatabase bank_database)
	{
		super(account_number, screen, bank_database);
		this.keypad = keypad;
	}
	
	private int depositAmount()
	{
		Screen screen = getScreen();
		screen.displayMessage("\n입금액을 입력하시오 (취소하려면 0): ");
		int input = keypad.getInteger();
		return (input == CANCELED) ? CANCELED : input;
	}
	
	public void execute()
	{
		BankDatabase bank_database = getBankDatabase();
		Screen screen = getScreen();
		amount = depositAmount();
		if(amount != CANCELED)
		{
			bank_database.credit(getAccountNumber(), amount);
			screen.displayMsgLine("\n입금이 완료되었습니다.");
		}
		else
		{
			screen.displayMsgLine("\n입금을 취소합니다.");
		}
	}
}

File: kumoh-code-2017/C++/#2 설계/과제15 은행관리 프로그램/Deposit.cpp
#include"Deposit.h"

Deposit::Deposit(int accountNumber, Screen screen, BankDatabase bankDB,Keypad keypad) : Transaction( accountNumber,  screen,  bankDB){
	this->keypad = keypad;
}

int Deposit::getDepositAmountFromKBD(){
	Screen screen = getScreen();
	screen.displayMsg("입금액을 입력하시오(취소하려면 -999): ");
	int balance = keypad.getInput();

	if (balance != stop)
		return balance;
	else
		return NULL;
}

void Deposit::execute(){
	BankDatabase bankDatabse = getBankDataBalance();
	Screen screen = getScreen();
	this->money = getDepositAmountFromKBD();
	bankDatabse.credit(getAccountNumber(), bankDatabse.setBalance(money));// setbalance를 통해 간단히 표현(다형성의 장점)
}